### ISSUE TITLE
Support RFC-7807 through schemas for `problemdetails` or  `http-api-problem`

### ISSUE THREAD

**Issue by mettekou** [juhaku/utoipa#590]
[RFC-7807](https://www.rfc-editor.org/rfc/rfc7807) is a popular standard for HTTP API error response bodies. Both the [`problemdetails`](https://docs.rs/problemdetails/latest/problemdetails/) and [`http-api-problem`](https://docs.rs/http-api-problem/0.56.0/http_api_problem/) implement this standard. Would it be a good idea to implement schemas for the relevant types from these crates?

**Comment by juhaku** [comment]
Sure why not. The possibilities of how to integrate / implement such capability to `utoipa` can be explored in future. Though it might be better if the support is directly implemented to the corresponding crates, so whenever they change the changes are reflected to `utoipa` as well. Otherwise `utoipa` must be able to keep track of possible changes in the spec and act accordingly. However if, and most likely so, the spec is not going to be changed frequently which would help to keep the maintenance overhead low. But the manual implementation of schemas would need duplicate work for declaring the properties of the possible types.

**Comment by mettekou** [comment]
> Sure why not. The possibilities of how to integrate / implement such capability to `utoipa` can be explored in future. Though it might be better if the support is directly implemented to the corresponding crates, so whenever they change the changes are reflected to `utoipa` as well. Otherwise `utoipa` must be able to keep track of possible changes in the spec and act accordingly. However if, and most likely so, the spec is not going to be changed frequently which would help to keep the maintenance overhead low. But the manual implementation of schemas would need duplicate work for declaring the properties of the possible types.

@juhaku So I had better contact the authors of those libraries about integration with `utoipa`, then?

**Comment by Drifter-J** [comment]
I would suggest closing this as there is workaround, @juhaku
I checked following issues:   
https://github.com/chridou/http-api-problem/issues/46
https://github.com/sazzer/problemdetails/issues/3

none of them were addressed.

**Comment by juhaku** [comment]
@Drifter-J yeah, seems like those crates solves a niche problem and are not that actively maintained nor widely adopted. Nonetheless can be closed now and labelled as `wontfix`. At the moment there is bandwidth to take on this matter and is not a focus of the `utoipa` project anyways. 

If somebody is willing to put the time and effort for supporting the`RFC-7807` this can be reevaluated then again.
